specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2023-08-01-preview/appconfiguration.json

```diff
@@ -0,0 +1,2890 @@
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "status": {
+          "description": "The current status of the snapshot.",
+          "enum": [
+            "Provisioning",
+            "Ready",
+            "Archived",
+            "Failed"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "SnapshotStatus",
+            "modelAsString": true
+          }
+        },
+        "filters": {
+          "description": "A list of filters used to filter the key-values included in the snapshot.",
+          "type": "array",
+          "minItems": 1,
+          "maxItems": 3,
+          "items": {
+            "$ref": "#/definitions/KeyValueFilter"
+          },
+          "x-ms-identifiers": []
+        },
+        "compositionType": {
+          "description": "The composition type describes how the key-values within the snapshot are composed. The 'key' composition type ensures there are no two key-values containing the same key. The 'key_label' composition type ensures there are no two key-values containing the same key and label.",
+          "type": "string",
+          "enum": [
+            "Key",
+            "Key_Label"
+          ],
+          "x-ms-enum": {
+            "name": "CompositionType",
+            "modelAsString": true
+          }
+        },
+        "created": {
+          "description": "The time that the snapshot was created.",
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "expires": {
+          "description": "The time that the snapshot will expire.",
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "retentionPeriod": {
+          "description": "The amount of time, in seconds, that a snapshot will remain in the archived state before expiring. This property is only writable during the creation of a snapshot. If not specified, the default lifetime of key-value revisions will be used.",
+          "type": "integer",
+          "minimum": 3600,
+          "maximum": 7776000,
+          "format": "int64",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "size": {
+          "description": "The size in bytes of the snapshot.",
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true
+        },
+        "itemsCount": {
+          "description": "The amount of key-values in the snapshot.",
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true
+        },
+        "tags": {
+          "description": "The tags of the snapshot.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "etag": {
```

Comment:
>etag [](http://example.com/codeflow?start=9&length=4)

This needs to go at the top level, or be renamed to something else. If this is a proxy resource, the best way to move this property to the top level is to use the AzureEntityResource definition.

Reply:
This is a proxy for a data plane snapshot and is modeled similarly to the existing KeyValue resource, which is a proxy for a data plane KeyValue with an etag in the properties too:

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/stable/2023-03-01/appconfiguration.json#L1803

The intent is to have all the data plane properties here, and I'm not following on why it would need to be renamed if it is here. Could you clarify?